src/controller/CommissioneeDeviceProxy.cpp

```diff
@@ -0,0 +1,329 @@
+/*
+ *
+ *    Copyright (c) 2020-2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *  @file
+ *    This file contains implementation of Device class. The objects of this
+ *    class will be used by Controller applications to interact with CHIP
+ *    devices. The class provides mechanism to construct, send and receive
+ *    messages to and from the corresponding CHIP devices.
+ */
+
+#include <controller/CommissioneeDeviceProxy.h>
+
+#include <controller-clusters/zap-generated/CHIPClusters.h>
+
+#include <app/CommandSender.h>
+#include <app/ReadPrepareParams.h>
+#include <app/util/DataModelHandler.h>
+#include <lib/core/CHIPCore.h>
+#include <lib/core/CHIPEncoding.h>
+#include <lib/core/CHIPSafeCasts.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/ErrorStr.h>
+#include <lib/support/SafeInt.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+using namespace chip::Callback;
+
+namespace chip {
+
+CHIP_ERROR CommissioneeDeviceProxy::LoadSecureSessionParametersIfNeeded(bool & didLoad)
+{
+    didLoad = false;
+
+    // If there is no secure connection to the device, try establishing it
+    if (mState != ConnectionState::SecureConnected)
```

Comment:
This part of the logic is mostly copied from CHIPDevice.cpp. I honestly think that it needs some streamlining, but may be out of scope of this PR.

The main reason for `LoadSecureSessionParametersIfNeeded` was to do on demand loading of secure session from the stored device object state. With this refactor, the CommissioneeDeviceProxy will only use PASE based secure session, and that will never be persisted in the storage. So once the device has established the PASE session, it'll always be in the connected state, until someone releases the object.

I can probably add a TODO to this code so that it can be cleaned up.